exchange/exchange_test.go

```diff
@@ -1196,6 +1201,131 @@ func TestCategoryDedupe(t *testing.T) {
+
+	seatBid = pbsOrtbSeatBid{innerBids, "USD", nil, nil}
+
+	bidCategory, adapterBids, rejections, err = applyCategoryMapping(nil, requestExt, adapterBids, categoriesFetcher, targData)
+
+	assert.Equal(t, nil, err, "Category mapping error should be empty")
+	assert.Equal(t, 1, len(rejections), "There should be 1 bid rejection message")
+	assert.Equal(t, "bid rejected [bid ID: bid_id1] reason: Bid duration exceeds maximum allowed", rejections[0], "Error message did not match expected result")
+	assert.Equal(t, 0, len(adapterBids[bidderName].bids), "Bidders number doesn't match")
+	assert.Equal(t, 0, len(bidCategory), "Bidders category mapping doesn't match")
+
+	cats1 := []string{"IAB1-3"}
+	cats2 := []string{"IAB1-3"}
+	cats3 := []string{"IAB1-4"}
+	bid1 := openrtb.Bid{ID: "bid_id1", ImpID: "imp_id1", Price: 10.0000, Cat: cats1, W: 1, H: 1}
+	bid2 := openrtb.Bid{ID: "bid_id2", ImpID: "imp_id2", Price: 10.0000, Cat: cats2, W: 1, H: 1}
+	bid3 := openrtb.Bid{ID: "bid_id3", ImpID: "imp_id3", Price: 10.0000, Cat: cats3, W: 1, H: 1}
+
+	bid1_1 = pbsOrtbBid{&bid1, "video", nil, &openrtb_ext.ExtBidPrebidVideo{Duration: 30}}
+	bid1_2 := pbsOrtbBid{&bid2, "video", nil, &openrtb_ext.ExtBidPrebidVideo{Duration: 30}}
+	bid1_3 := pbsOrtbBid{&bid3, "video", nil, &openrtb_ext.ExtBidPrebidVideo{Duration: 30}}
+
+	innerBids = []*pbsOrtbBid{
+		&bid1_1,
+		&bid1_2,
+		&bid1_3,
+	}
+
+	seatBid = pbsOrtbSeatBid{innerBids, "USD", nil, nil}
+
+	bidCategory, adapterBids, rejections, err = applyCategoryMapping(nil, requestExt, adapterBids, categoriesFetcher, targData)
+
+	// There is a random factor to deduplication
+	// Determine which bid was accepted/rejected by checking the actual result
+	accepted, rejected := "", ""
+	if _, ok := bidCategory["bid_id1"]; ok {
+		accepted = "bid_id1"
+		rejected = "bid_id2"
+	} else {
+		accepted = "bid_id2"
+		rejected = "bid_id1"
+	}
+
+	assert.Equal(t, nil, err, "Category mapping error should be empty")
+	assert.Equal(t, 1, len(rejections), "There should be 1 bid rejection message")
+	assert.Equal(t, fmt.Sprintf("bid rejected [bid ID: %s] reason: Bid was deduplicated", rejected), rejections[0], "Error message did not match expected result")
+	assert.Equal(t, "", bidCategory[rejected], "Bid should have been removed")
+	assert.Equal(t, "10.00_Electronics_30s", bidCategory[accepted], "Category mapping doesn't match")
+	assert.Equal(t, "10.00_Sports_30s", bidCategory["bid_id3"], "Category mapping doesn't match")
+	assert.Equal(t, 2, len(adapterBids[bidderName].bids), "Bidders number doesn't match")
+	assert.Equal(t, 2, len(bidCategory), "Bidders category mapping doesn't match")
+}
+
+func TestUpdateRejections(t *testing.T) {
+	rejections := []string{}
+
+	rejections = updateRejections(rejections, "bid_id1", "some reason 1")
+	rejections = updateRejections(rejections, "bid_id2", "some reason 2")
+
+	assert.Equal(t, 2, len(rejections), "Rejections should contain 2 rejection messages")
```

Comment:
I don't think the order of the rejection messages matter, so consider using `assert.Contains`.

Reply:
Updated to use `assert.Contains` instead